test/list.go

```diff
@@ -486,6 +489,130 @@ func DescribeListTestsFor(ctx *common.TestContext, t TestCase, responseMode Resp
 				})
 			}
 
+			if t.SupportsAsyncOperations {
+				Context("when attach_last_operations is truthy", func() {
+					var resourceWithOneOperation, resourceWithFewOperations common.Object
+					var lastOperationForResourceId string
+					var resourceWithOneOperationId, resourceWithFewOperationsId string
+
+					createTestOperation := func(resourceID string, opType types.OperationCategory) (types.Object, string) {
+						id, err := uuid.NewV4()
+						Expect(err).ToNot(HaveOccurred())
+						labels := make(map[string][]string)
+						testResource, err := ctx.SMRepository.Create(context.TODO(), &types.Operation{
+							Base: types.Base{
+								ID:        id.String(),
+								CreatedAt: time.Now(),
+								UpdatedAt: time.Now(),
+								Labels:    labels,
+								Ready:     true,
+							},
+							Description:   "test",
+							Type:          opType,
+							State:         types.SUCCEEDED,
+							ResourceID:    resourceID,
+							ResourceType:  types.ObjectType(t.API),
+							CorrelationID: id.String(),
+						})
+						Expect(err).ShouldNot(HaveOccurred())
+						return testResource, id.String()
+					}
+
+					BeforeEach(func() {
+						resourceWithOneOperation = t.ResourceBlueprint(ctx, ctx.SMWithOAuth, bool(responseMode))
+						resourceWithFewOperations = t.ResourceBlueprint(ctx, ctx.SMWithOAuth, bool(responseMode))
+						resourceWithOneOperationId = resourceWithOneOperation["id"].(string)
+						resourceWithFewOperationsId = resourceWithFewOperations["id"].(string)
+						createTestOperation(resourceWithFewOperationsId, types.CREATE)
+						_, lastOperationForResourceId = createTestOperation(resourceWithFewOperationsId, types.DELETE)
+					})
+
+					AfterEach(func() {
+						err := common.RemoveResourceByCriterion(ctx, query.ByField(query.InOperator, "resource_id", []string{resourceWithOneOperationId, resourceWithOneOperationId}...), types.OperationType)
+						Expect(err).ShouldNot(HaveOccurred())
+						err = common.RemoveResourceByCriterion(ctx, query.ByField(query.InOperator, "id", []string{resourceWithOneOperationId, resourceWithOneOperationId}...), t.ResourceType)
+						if err != nil {
+							print(err.Error())
+						}
+						Expect(err).ShouldNot(HaveOccurred())
+					})
+
+					It("retrieves the resources list when each resource contains the last operation it is associated to", func() {
+						resp := ctx.SMWithOAuth.GET(t.API).
+							WithQuery("attach_last_operations", "true").
+							Expect().Status(http.StatusOK).
+							JSON()
+
+						for _, resource := range resp.Path("$.items[*]").Array().Iter() {
+							itemResourceId := resource.Object().Value("id").String().Raw()
+							if itemResourceId == resourceWithOneOperationId || itemResourceId == resourceWithFewOperationsId {
```

Comment:
are any other resources expected? if it is always just those 2 the if can be removed

Reply:
yes.. this test is going to be executed by different tests that prepare content that is not cleaned. actually we can't assume what resources will be stored in the database.